File: pi.java
public class pi {
  static double myMethod(long n) {
    double pi = 0.0;
    int sign = 1;

    for(long i=0; i<n; i++){
        pi += sign * 1.0 / (2*i + 1);
        sign = -sign;
    }
    pi *= 4;
    return pi;
}
  

  public static void main(String[] args) {
    System.out.println(myMethod(2000000000));
  }
}

File: pi.rb
def pi_approx(n)
  pi = 0.0
  sign = 1
  n.times do |i|
    pi += sign * 1.0 / (2 * i + 1)
    sign = -sign
  end
  pi * 4
end

puts pi_approx(2000000000)